Sources/DataPipeline/DataPipelineImplementation.swift

```diff
@@ -252,3 +251,88 @@ extension DataPipelineImplementation {
+        }
+        if let traits = body {
+            let jsonTraits = try? JSON(traits)
+            identifyEvent.traits = jsonTraits
+        }
+        analytics.process(event: identifyEvent)
+    }
+
+    func processScreenEventFromBGQ(identifier: String, name: String, timestamp: String?, properties: [String: Any]) {
+        var screenEvent = ScreenEvent(title: name, category: nil)
+        screenEvent.userId = identifier
+        screenEvent.timestamp = timestamp
+        if let jsonProperties = try? JSON(properties) {
+            screenEvent.properties = jsonProperties
+        }
+        analytics.process(event: screenEvent)
+    }
+
+    func processEventFromBGQ(identifier: String, name: String, timestamp: String?, properties: [String: Any]) {
+        var trackEvent = TrackEvent(event: name, properties: nil)
+        trackEvent.userId = identifier
+        trackEvent.timestamp = timestamp
+        if let jsonProperties = try? JSON(properties) {
+            trackEvent.properties = jsonProperties
+        }
+        analytics.process(event: trackEvent)
+    }
+
+    func processDeleteTokenFromBGQ(identifier: String, token: String, timestamp: String) {
+        var trackDeleteEvent = TrackEvent(event: "Device Deleted", properties: nil)
+        trackDeleteEvent.userId = identifier
+        trackDeleteEvent.timestamp = timestamp
+        let journeyDict: [String: Any] = ["journeys": ["identifiers": ["id": identifier]]]
+        let deviceDict: [String: Any] = ["device": ["token": token, "type": "ios"]]
+        if let context = try? JSON(deviceDict.mergeWith(journeyDict)) {
+            trackDeleteEvent.context = context
+        }
+        analytics.process(event: trackDeleteEvent)
+    }
+
+    func processRegisterDeviceFromBGQ(identifier: String, token: String, timestamp: String, attributes: [String: Any]? = nil) {
+        var trackRegisterTokenEvent = TrackEvent(event: "Device Created or Updated", properties: nil)
+        trackRegisterTokenEvent.userId = identifier
+        trackRegisterTokenEvent.timestamp = timestamp
+        let journeyDict: [String: Any] = ["journeys": ["identifiers": ["id": identifier]]]
+        var tokenDict: [String: Any] = ["token": token, "type": "ios"]
+
+        if let attributes = attributes {
+            tokenDict = tokenDict.mergeWith(attributes)
+        }
+
+        let deviceDict: [String: Any] = ["device": tokenDict]
+        if let context = try? JSON(deviceDict.mergeWith(journeyDict)) {
+            trackRegisterTokenEvent.context = context
+        }
+        analytics.process(event: trackRegisterTokenEvent)
+    }
+
+    func processPushMetricsFromBGQ(token: String, event: Metric, deliveryId: String, timestamp: String, metaData: [String: Any] = [:]) {
+        let properties: [String: Any] = metaData.mergeWith([
+            "metric": event.rawValue,
+            "deliveryId": deliveryId,
+            "recipient": token
+        ])
+//        if let token = deviceToken {
```

Comment:
why do we have this commented-out version here?

Reply:
My bad ! I made this change to test a condition and didn't reverse it.